File: src/DMS.Abstraction/Documents/Document.cs
using DMS.Abstraction.Revisions;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DMS.Abstraction
{
    [BsonIgnoreExtraElements]
    public class Document : IDocument
    {
        //public Document()
        //{
        //    Versions = new List<IVersion>();
        //    Revisions = new List<IRevision>();
        //}
        //public List<IVersion> Versions { get; set; }

        //public List<IRevision> Revisions { get; set; }
        public int CreatedBy { get; set; }

        public DateTime CreatedOn { get; set; }

        public int CurrentRevision { get; set; }

        public int CurrentVersion { get; set; }

        public int? DeletedBy { get; set; }

        public DateTime DeletedOn { get; set; }

        public int DocumentId { get; set; }

        public string Extension { get; set; }

        public string FileName { get; set; }

        public bool IsDeleted { get; set; }

        public bool IsShared { get; set; }

        public int? LockedBy { get; set; }

        public string ModifiedBy { get; set; }

        public DateTime ModifiedOn { get; set; }

        public byte[] DocumentData { get; set; }

        public string DocumentTags { get; set; }

        public string CreatedByName { get; set; }

        public string LockedByName { get; set; }
    }
}

File: src/DMS.WebAppUI/src/app/document/document.ts

/* Defines the IDocument entity */
export interface IDocument {
    CreatedByName: string;
    CreatedOn: string;
    CurrentRevision: number;
    CurrentVersion: number;
    DeletedBy: string;
    DeletedOn: string;
    DocumentId: number;
    Extension: string;
    FileName: string;
    IsDeleted: boolean;
    IsShared: boolean;
    LockedByName: string;
    ModifiedBy: string;
    ModifiedOn: string;
    DocumentData: any;
    DocumentTags: string;
    LockStatus: string;
    IsDocCheckedOut: boolean;
}

export class Document implements IDocument {
    constructor(
        public CreatedByName: string,
        public CreatedOn: string,
        public CurrentRevision: number,
        public CurrentVersion: number,
        public DeletedBy: string,
        public DeletedOn: string,
        public DocumentId: number,
        public Extension: string,
        public FileName: string,
        public IsDeleted: boolean,
        public IsShared: boolean,
        public LockedByName: string,
        public ModifiedBy: string,
        public ModifiedOn: string,
        public DocumentData: any,
        public DocumentTags: string,
        public LockStatus: string,
        public IsDocCheckedOut: boolean
    ) { }

}

export interface IAdvanceSearch {
    FileName: string;
    Extension: string;  
    FromDate: Date;
    ToDate: Date;
}
export class AdvanceSearch implements IAdvanceSearch {
    constructor(
        public FileName: string,
        public Extension: string,
        public FromDate: Date,
        public ToDate : Date
    ) { }
}

export enum DocType {
    Personal,
    Public
}